subsys/bluetooth/mesh/Kconfig

```diff
@@ -37,7 +37,7 @@ config BT_MESH_GATT_CLIENT
 config BT_MESH_GATT_SERVER
 	bool
 	select BT_MESH_GATT
-	select BT_GATT_DYNAMIC_DB
+	depends on BT_GATT_DYNAMIC_DB
```

Comment:
Does not changing this generate build warnings/errors?

Reply:
Yes, it leads to build errors related to the dependency loop, and it is one of the easiest ways to solve it.